diff --git a/target/linux/ipq806x/dts/qcom-ipq8068-ap-32x.dts b/target/linux/ipq806x/dts/qcom-ipq8068-ap-32x.dts
index be50a44..f927889 100644
--- a/target/linux/ipq806x/dts/qcom-ipq8068-ap-32x.dts
+++ b/target/linux/ipq806x/dts/qcom-ipq8068-ap-32x.dts
@@ -19,15 +19,15 @@
 		serial0 = &gsbi4_serial;
 		serial1 = &gsbi2_serial;
 
-		ethernet0 = &gmac2;
-		ethernet1 = &gmac3;
+		ethernet0 = &gmac3;
 
 		led-boot = &led_power_amber;
 		led-failsafe = &led_power_red;
 		led-running = &led_power_green;
 		led-upgrade = &led_power_amber;
 
-		label-mac-device = &gmac2;
+		label-mac-device = &gmac3;
+		mdio-gpio0 = &mdio_gpio0;
 	};
 
 	chosen {
@@ -87,6 +87,23 @@
 		};
 	};
 
+	mdio_gpio0: mdio {
+		compatible = "virtual,mdio-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		gpios = <&qcom_pinmux 1 GPIO_ACTIVE_HIGH>,
+			<&qcom_pinmux 0 GPIO_ACTIVE_HIGH>;
+		pinctrl-0 = <&mdio_gpio_pins>;
+		pinctrl-names = "default";
+
+		phy5: ethernet-phy@5 {
+			reg = <5>;
+			reset-gpios = <&qcom_pinmux 67 GPIO_ACTIVE_LOW>;
+			reset-assert-us = <10000>;
+			reset-deassert-us = <10000>;
+		};
+	};
+
 	watchdog {
 		compatible = "linux,wdt-gpio";
 		gpios = <&qcom_pinmux 61 GPIO_ACTIVE_LOW>;
@@ -150,6 +167,15 @@
 		};
 	};
 
+	mdio_gpio_pins: mdio_gpio_pins {
+		mux {
+			pins = "gpio0", "gpio1";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-disable;
+		};
+	};
+
 	watchdog_pins: watchdog_pins {
 		mux {
 			pins = "gpio61";
@@ -304,30 +330,11 @@
 };
 
 &mdio0 {
-	status = "okay";
-
-	pinctrl-0 = <&mdio0_pins>;
-	pinctrl-names = "default";
-
-	phy0: ethernet-phy@0 {
-		reg = <0>;
-	};
-
-	phy1: ethernet-phy@1 {
-		reg = <1>;
-	};
+	status = "disabled";
 };
 
 &gmac2 {
-	status = "okay";
-
-	qcom,id = <2>;
-
-	phy-mode = "sgmii";
-	phy-handle = <&phy0>;
-
-	nvmem-cells = <&macaddr_lan 0>;
-	nvmem-cell-names = "mac-address";
+	status = "disabled";
 };
 
 &gmac3 {
@@ -336,9 +343,9 @@
 	qcom,id = <3>;
 
 	phy-mode = "sgmii";
-	phy-handle = <&phy1>;
+	phy-handle = <&phy5>;
 
-	nvmem-cells = <&macaddr_lan 1>;
+	nvmem-cells = <&macaddr_lan 0>;
 	nvmem-cell-names = "mac-address";
 };
 

[executed on device: server1 (b9b9f4cb-a4ae-475c-90b6-cd67f8dc636f)]